src/components/checkboxTree/components/RenderComposition.tsx

```diff
@@ -46,6 +47,14 @@ export function RenderComposition({
   };
 
   React.useEffect(() => {
```

Comment:
Question:
What's the reason to not specify useEffect dependencies? It does not cause any problems?

Reply:
I tested it now, it doesn't cause any problems, but providing at least `selectedList` (the only dynamic part) as a dependency, seems to cut a few not needed re-renders. Adding all missed deps.